sklearn/compose/_target.py

```diff
@@ -173,9 +174,18 @@ def _fit_transformer(self, y):
         elif self.transformer is not None:
             self.transformer_ = clone(self.transformer)
         else:
-            if self.func is not None and self.inverse_func is None:
+            if (self.func is not None and self.inverse_func is None) or (
+                self.inverse_func is not None and self.func is None
+            ):
```

Comment:
Very nit: I think keeping the order between the two lines make it easier to read:

```suggestion
            if (self.func is not None and self.inverse_func is None) or (
                self.func is None and self.inverse_func is not None
            ):
```